In [1]:
# Import libraries
import math

***Goal: to approximate and show the identity for the Gaussian integral (from negative infinity to positive infinity) being equal to the square root of pi.***

We know the Riemann Sum is defined as:  
$$
\sum_{i=1}^{n} f(x_i) \Delta x
$$

And we can use it because of its relationship to an integration, which follows:  
$$
\lim_{n \to \infty} \sum_{i=1}^{n} f(x_i) \Delta x = \int_a^b f(x) \, dx
$$

The Gaussian function is:  
$$
e^{-x^2}
$$
and we want to simulate:  
$$
\int e^{-x^2} \,dx
$$

The Gaussian function is: $$e^{-x^2}$$ and we want to simulate: $$\int e^{-x^2} \,dx$$


In [2]:
# Make a function that returns the y value to the Gaussian function
def gaussian_function(x):
    return math.e**(-x**2)

In [3]:
# Test some values to ensure functionality
print("f(1) =", gaussian_function(1))
print("f(-1) =", gaussian_function(-1))
print("f(0.5) =", gaussian_function(0.5))
print("f(0) =", gaussian_function(0))

f(1) = 0.36787944117144233
f(-1) = 0.36787944117144233
f(0.5) = 0.7788007830714049
f(0) = 1.0


We know the Riemann Sum is defined as: $$\sum_{i=1}^{n} f(x_i) \Delta x$$

And we can use it because of its relationship to an integration, which follows: $$\lim_{n \to \infty} \sum_{i=1}^{n} f(x_i) \Delta x = \int_a^b f(x) \, dx$$

In [4]:
# Define and create Riemann summation
start = -100
end = 100
simulated_vals = range(start, end)
delta = 0.1
summation = 0


while start < end:
    summation += delta * gaussian_function(start)
    start += delta

print("Summation:",  summation)

Summation: 1.7724538509055157


$$1.7724538509055157 \approx \sqrt{\pi}, \text{ and thus by simulation we observe the identity stands.}$$